src/cr/cube/dimension.py

```diff
@@ -19,124 +21,125 @@ class Dimension(object):
     :attr:`.CrunchCube.dimensions`.
     """
 
-    def __init__(self, dim, selections=None):
-        self._dim = dim
-        self._type = self._get_type(dim, selections)
+    def __init__(self, dimension_dict, next_dimension_dict=None):
+        self._dimension_dict = dimension_dict
+        self._next_dimension_dict = next_dimension_dict
 
     @lazyproperty
     def alias(self):
-        """Alias of a cube's dimension."""
-        refs = self._dim['references']
+        """str system (as opposed to human) name for this dimension."""
+        refs = self._dimension_dict['references']
         return refs.get('alias')
 
     @lazyproperty
     def description(self):
-        """Description of a cube's dimension."""
-        refs = self._dim['references']
-        return refs.get('description')
+        """str description of this dimension."""
+        description = self._dimension_dict['references'].get('description')
+        return description if description else ''
+
+    @lazyproperty
+    def dimension_type(self):
```

Comment:
This is the refactored form of `.type`, `._type`, `_get_type()`, and `._is_multiple_response()`. Calling `._get_type()` in the constructor represented "real work", which is why you had to mock it so often in unit tests even when it didn't have anything to do with that unit. The classmethods were unnecessary once `next_dimension_dict` was stored as an instance var and `.dimension_type` was made lazy. All this logic really belongs in a `Dimensions` collection object anyway, which we'll probably get to before too long.